stout/flags.cc

```
@@ -18,63 +18,185 @@ namespace stout::flags {
+          }
 
-    if (flag.names().empty()) {
-      std::cerr
-          << "Missing at least one flag name in 'names' for field '"
-          << field->full_name() << "'"
-          << std::endl;
-      std::exit(1);
-    }
+          if (subcommand.help().empty()) {
+            std::cerr
+                << "Missing subcommand 'help' for field '"
+                << field->full_name() << "'"
+                << std::endl;
+            std::exit(1);
+          }
+
+          for (const auto& name : subcommand.names()) {
+            auto [_, inserted] = subcommand_fields_.emplace(name, field);
+
+            if (!inserted) {
+              std::cerr << "Encountered duplicate subcommand name "
+                        << "'" << name << "' for message '"
+                        << message->GetTypeName() << "'"
+                        << std::endl;
+              std::exit(1);
+            }
+          }
+
+          for (const auto& name : subcommand.deprecated_names()) {
+            auto [_, inserted] = subcommand_fields_.emplace(name, field);
+
+            if (!inserted) {
+              std::cerr << "Encountered duplicate (deprecated) subcommand name "
+                        << "'" << name << "' for message '"
+                        << message->GetTypeName() << "'"
+                        << std::endl;
+              std::exit(1);
+            }
+          }
+
+          // Create nested parser and store it to the 'nested_parsers_'.
+          auto [_, inserted] = nested_parsers_.emplace(
+              field,
+              std::make_unique<Parser>(
+                  Parser::Builder(
+                      message->GetReflection()->MutableMessage(message, field),
+                      true)
+                      .Build()));
 
-    if (flag.help().empty()) {
-      std::cerr
-          << "Missing flag 'help' for field '" << field->full_name() << "'"
-          << std::endl;
-      std::exit(1);
-    }
+          CHECK(inserted);
```

I would add some debug log for this:)
```suggestion
          CHECK(inserted) << "Failed to emplace the new nested parser for 'nested_parsers_'";
```

---

This is an internal invariant rather than an external once where messages printed out might be helpful. In other words, if this `CHECK` fails we need to fix the code, our users can't change something to make it work. So I'm ok leaving these out. If we need/want our users to do something else we should return an error, or exit (which is what we do), or last case scenario add a `CHECK()` for a precondition that they violated. There isn't a precondition here.